src/app/clusters/temperature-measurement-server/temperature-measurement-server.cpp

```diff
@@ -15,78 +15,95 @@
- *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- *    See the License for the specific language governing permissions and
- *    limitations under the License.
- */
-/****************************************************************************
- * @file
- * @brief Routines for the Temperature Measurement Server
- *plugin.
- *******************************************************************************
- ******************************************************************************/
-
 #include "temperature-measurement-server.h"
 
 #include <app/util/af.h>
 
-#include <app/util/af-event.h>
-#include <app/util/attribute-storage.h>
-
+#include "gen/att-storage.h"
 #include "gen/attribute-id.h"
 #include "gen/attribute-type.h"
 #include "gen/cluster-id.h"
+#include "gen/command-id.h"
 
-using namespace chip;
+#include <support/logging/CHIPLogging.h>
 
-EmberEventControl emberAfPluginTemperatureMeasurementServerReadEventControl;
+EmberAfStatus emberAfTemperatureMeasurementClusterGetMeasuredValue(chip::EndpointId endpoint, int16_t * measuredValue)
+{
+    return emberAfReadAttribute(endpoint, ZCL_TEMP_MEASUREMENT_CLUSTER_ID, ZCL_TEMP_MEASURED_VALUE_ATTRIBUTE_ID,
+                                CLUSTER_MASK_SERVER, (uint8_t *) measuredValue, sizeof(*measuredValue), NULL);
+}
 
-// TODO: There's no header that declares this event handler, and it's not 100%
-// clear where best to declare it.
-// https://github.com/project-chip/connectedhomeip/issues/3619
-void emberAfPluginTemperatureMeasurementServerReadEventHandler() {}
+EmberAfStatus emberAfTemperatureMeasurementClusterGetMinMeasuredValue(chip::EndpointId endpoint, int16_t * minMeasuredValue)
+{
+    return emberAfReadAttribute(endpoint, ZCL_TEMP_MEASUREMENT_CLUSTER_ID, ZCL_TEMP_MIN_MEASURED_VALUE_ATTRIBUTE_ID,
+                                CLUSTER_MASK_SERVER, (uint8_t *) minMeasuredValue, sizeof(*minMeasuredValue), NULL);
+}
 
-void emberAfPluginTemperatureMeasurementServerStackStatusCallback(EmberStatus status) {}
+EmberAfStatus emberAfTemperatureMeasurementClusterGetMaxMeasuredValue(chip::EndpointId endpoint, int16_t * maxMeasuredValue)
+{
+    return emberAfReadAttribute(endpoint, ZCL_TEMP_MEASUREMENT_CLUSTER_ID, ZCL_TEMP_MAX_MEASURED_VALUE_ATTRIBUTE_ID,
+                                CLUSTER_MASK_SERVER, (uint8_t *) maxMeasuredValue, sizeof(*maxMeasuredValue), NULL);
+}
 
-// -------------------------------------------------------------------------
-// ****** callback section *******
+void emberAfTemperatureMeasurementClusterServerInitCallback(chip::EndpointId endpoint)
+{
+    int16_t value        = 0;
+    EmberAfStatus status = emberAfTemperatureMeasurementClusterGetMeasuredValue(endpoint, &value);
 
-void emberAfPluginTemperatureMeasurementServerInitCallback(void)
+    if (EMBER_ZCL_STATUS_SUCCESS != status)
+    {
+        (void) emberAfTemperatureMeasurementClusterSetMeasuredValueCallback(endpoint, 0);
```

Comment:
It think it is worthwhile to check the return value of `emberAfTemperatureMeasurementClusterSetMeasuredValueCallback` and log a message using `ChipLogError` if something went wrong.
Same comment for  `emberAfTemperatureMeasurementClusterSetMinMeasuredValueCallback` and `emberAfTemperatureMeasurementClusterSetMaxMeasuredValueCallback`

Reply:
If I enable the `emberAfTemperatureMeasurementClusterPrintln` in the `emberAfTemperatureMeasurementClusterSetMeasuredValueCallback` it will not give much more insides checking the return value. So I will enable it in the function itself.

Reply:
Ok, I will take the comments you provided and let the caller do the logging. So I will use the return value in the init function.